pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationManager.java

```diff
@@ -78,14 +79,54 @@ public boolean canProduce(DestinationName destination, String role) throws Excep
      *            the fully qualified destination name associated with the destination.
      * @param role
      *            the app id used to receive messages from the destination.
+     * @param subscription
+     *            the subscription name defined by the client
      */
+    public CompletableFuture<Boolean> canConsumeAsync(DestinationName destination, String role, String subscription) {
+        CompletableFuture<Boolean> permissionFuture = new CompletableFuture<>();
+        try {
+            configCache.policiesCache().getAsync(POLICY_ROOT + destination.getNamespace()).thenAccept(policies -> {
+                if (!policies.isPresent()) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Policies node couldn't be found for destination : {}", destination);
+                    }
+                } else {
+                    if (isNotBlank(subscription)) {
+                        switch (policies.get().subscription_auth_mode) {
+                        case Prefix:
+                            if (!subscription.startsWith(role)) {
+                                permissionFuture.complete(false);
+                                return;
```

Comment:
Should we return `permissionFuture` here?
```
return permissionFuture;
```

Reply:
This `return` is for lambda of [line 88](https://github.com/apache/incubator-pulsar/pull/899/files/3668902b56633635aa8b8b9ea3923d65970c5cbf#diff-0146488122772630082224995b59a1b1R88) and it need not return any values.

Reply:
I got it, thanks.